includes/payment-methods/class-wc-stripe-express-checkout-helper.php

```diff
@@ -1602,7 +1665,16 @@ public function is_amazon_pay_enabled() {
 	 * @return boolean
 	 */
 	public function is_link_enabled() {
-		return WC_Stripe_UPE_Payment_Method_Link::is_link_enabled( $this->gateway );
+		$is_enabled = WC_Stripe_UPE_Payment_Method_Link::is_link_enabled( $this->gateway );
+		if ( is_product() ) {
```

Comment:
Thought: do we want our own helper for `is_product()` to mirror what we're doing for `is_cart()` and `is_checkout()`? Maybe something for a follow-up PR -- I don't think we should tackle that in this PR.